string action
string model
string props
---
bool result
string err